RIOT-OS/RIOT
# sys/net/coap: cast in non-config define

### Contribution description

If we try to set the compile-time config we need to always remember casting, instead, just cast in the places that are not exposed to users.

### Testing procedure

In master run:
```
CFLAGS="-DCONFIG_COAP_ACK_TIMEOUT_MS=2000 -DCONFIG_COAP_RANDOM_FACTOR_1000=1500"  BOARD=atmega256rfr2-xpro make -C tests/net/gcoap_dns/
```

it should fail with:
<details>

```
/home/weiss/repos/RIOT/sys/net/application_layer/gcoap/gcoap.c: In function ‘_on_resp_timeout’:
/home/weiss/repos/RIOT/sys/net/application_layer/gcoap/gcoap.c:53:55: error: integer overflow in expression [-Werror=overflow]
 #define TIMEOUT_RANGE_END (CONFIG_COAP_ACK_TIMEOUT_MS * CONFIG_COAP_RANDOM_FACTOR_1000 / 1000)
                                                       ^
/home/weiss/repos/RIOT/sys/net/application_layer/gcoap/gcoap.c:596:34: note: in expansion of macro ‘TIMEOUT_RANGE_END’
         uint32_t end = (uint32_t)TIMEOUT_RANGE_END << i;
                                  ^
/home/weiss/repos/RIOT/sys/net/application_layer/gcoap/gcoap.c: In function ‘gcoap_req_send_tl’:
/home/weiss/repos/RIOT/sys/net/application_layer/gcoap/gcoap.c:53:55: error: integer overflow in expression [-Werror=overflow]
 #define TIMEOUT_RANGE_END (CONFIG_COAP_ACK_TIMEOUT_MS * CONFIG_COAP_RANDOM_FACTOR_1000 / 1000)
                                                       ^
/home/weiss/repos/RIOT/sys/net/application_layer/gcoap/gcoap.c:1682:56: note: in expansion of macro ‘TIMEOUT_RANGE_END’
                 timeout = random_uint32_range(timeout, TIMEOUT_RANGE_END);
                                                        ^
cc1: all warnings being treated as errors
make[3]: *** [/home/weiss/repos/RIOT/Makefile.base:143: /home/weiss/repos/RIOT/tests/net/gcoap_dns/bin/atmega256rfr2-xpro/gcoap/gcoap.o] Error 1
make[2]: *** [/home/weiss/repos/RIOT/Makefile.base:31: ALL--/home/weiss/repos/RIOT/sys/net/application_layer/gcoap] Error 2
make[1]: *** [/home/weiss/repos/RIOT/Makefile.base:31: ALL--/home/weiss/repos/RIOT/sys] Error 2
make: *** [/home/weiss/repos/RIOT/tests/net/gcoap_dns/../../../Makefile.include:769: application_tests_gcoap_dns.module] Error 2
make: Leaving directory '/home/weiss/repos/RIOT/tests/net/gcoap_dns'
```
</details>

With PR should fix it

### Issues/PRs references
